### PR TITLE
Add SearchApi Web Search Tool

### PR DESCRIPTION
**Pull request by raju249**
This PR adds [SearchApi](https://searchapi.io/) tool to perform web searches of various search engines like Google, Baidu, Bing, etc. It adds documentation to the Tools page to explain the usage. Based on our customer requests we are raising the PR to add SearchAPI as an option for the actions in OpenAGI.

SearchApi is ready to invest time in adding open-source integrations. Please feel free to mention @raju249 or @SebastjanPrachovskij for any future PRs related to SearchApi integration. We would be more than happy to help out.

### CODE REVIEW

**Comment by tarun-aiplanet** [comment]
Thank you for the PR, we will review it shortly.

File: docs/use-cases/jobsearch-agent.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ It utilize various tools for internet search and document comparison to fulfill
 
 First, we need to import the necessary modules. Each module serves a specific purpose in our script. We utilize various tools for internet search and document comparison to fulfill the agent's task. Here’s what each import does:
 
-* `GoogleSerpAPISearch` and `DuckDuckGoSearch` are tools for performing web searches.
+* `SearchApiSearch`, `GoogleSerpAPISearch` and `DuckDuckGoSearch` are tools for performing web searches.
```
Comment (tarun-aiplanet) [inline docs/use-cases/jobsearch-agent.md:9]: thank you for the PR. Could you please revert your changes for the use cases?

Rather please create a cookbook for showcasing the API

File: src/openagi/actions/tools/searchapi_search.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        }
+        logging.debug(f"{search_dict=}")
+        url = f"{base_url}?{urlencode(search_dict)}"
+        response = requests.request("GET", url)
+        json_response = response.json()
+
+        if not json_response:
+            raise OpenAGIException(f"Unable to generate result for the query {self.query}")
+
+        logging.debug(json_response)
```
Comment (tarun-aiplanet) [inline src/openagi/actions/tools/searchapi_search.py:36]: do comment this statement

**Comment by SebastjanPrachovskij** [comment]
Hey @tarun-aiplanet 

Thanks for the review!

I committed suggestions. Take a look whenever you can! :)

**Comment by tarun-aiplanet** [comment]
> Hey @tarun-aiplanet
> 
> Thanks for the review!
> 
> I committed suggestions. Take a look whenever you can! :)

there is a merge conflict, can you pull in the new changes

**Comment by SebastjanPrachovskij** [comment]
@tarun-aiplanet 

Solved conflicts.

**Comment by tarun-aiplanet** [comment]
> @tarun-aiplanet
> 
> Solved conflicts.

lgtm.

**Pull request merged**